polyglot/piranha/src/cleanup_rules/go/rules.toml

```diff
@@ -0,0 +1,547 @@
+groups = ["boolean_expression_simplify"]
+name = "simplify_something_and_true"
+query = """
+(
+    (binary_expression
+        left : (_)* @lhs
+        operator:"&&"
+        right: [(true) (parenthesized_expression (true))]
+    ) @binary_expression
+)
+"""
+replace = "@lhs"
+replace_node = "binary_expression"
+
+# Before :
+#  false && abc()
+# After :
+#  false
+#
+[[rules]]
+groups = ["boolean_expression_simplify"]
+name = "simplify_false_and_something"
+query = """
+(
+    (binary_expression
+        left: [(false) (parenthesized_expression (false))]
+        operator : "&&"
+        right : (_)* @rhs
+    ) @binary_expression
+)
+"""
+replace = "false"
+replace_node = "binary_expression"
+
+# Before :
+#  abc() && false
+# After :
+#  false
+#
+[[rules]]
+name = "simplify_something_and_false"
+groups = ["boolean_expression_simplify"]
+query = """
+(
+    (binary_expression
+        left : (_)* @lhs
+        operator : "&&"
+        right: [(false) (parenthesized_expression (false))]
+    ) @binary_expression
+)
+"""
+replace = "false"
+replace_node = "binary_expression"
+
+# Before :
+#  abc() || true
+# After :
+#  true
+#
+[[rules]]
+groups = ["boolean_expression_simplify"]
+name = "simplify_something_or_true"
+query = """
+(
+    (binary_expression
+        left : (_)* @lhs
+        operator:"||"
+        right: [(true) (parenthesized_expression (true))]
+    ) @binary_expression
+)
+"""
+replace = "true"
+replace_node = "binary_expression"
+
+# Before :
+#  true || abc()
+# After :
+#  true
+#
+[[rules]]
+groups = ["boolean_expression_simplify"]
+name = "simplify_true_or_something"
+query = """
+(
+    (binary_expression
+        left : [(true) (parenthesized_expression (true))]
+        operator:"||"
+        right: (_)* @rhs
+    ) @binary_expression
+)
+"""
+replace = "true"
+replace_node = "binary_expression"
+
+# Before:
+#  abc() || false
+# After :
+#  abc()
+#
+[[rules]]
+groups = ["boolean_expression_simplify"]
+name = "simplify_something_or_false"
+query = """
+(
+    (binary_expression
+        left : (_)* @lhs
+        operator:"||"
+        right: [(false) (parenthesized_expression (false))]
+    ) @binary_expression
+)
+"""
+replace = "@lhs"
+replace_node = "binary_expression"
+
+# Before:
+#  false || abc()
+# After :
+#  abc()
+#
+[[rules]]
+groups = ["boolean_expression_simplify"]
+name = "simplify_false_or_something"
+query = """
+(
+    (binary_expression
+        left : [(false) (parenthesized_expression (false))]
+        operator:"||"
+        right: (_)* @rhs
+    )
+@binary_expression)"""
+replace = "@rhs"
+replace_node = "binary_expression"
+
+# Simplifies equal identity comparison
+# Note that `nil == nil` is not compilable in Go, but compiles in tree-sitter
+#   true == true   -> true
+#   false == false -> true
+#   nil == nil     -> true
+#
+[[rules]]
+groups = ["boolean_expression_simplify"]
+name = "simplify_identity_equal"
+query = """
+(
+    (binary_expression
+        left: (
+            [
```

Comment:
What if we just use a wildcard for `@lhs` and `@rhs` ?

Reply:
Done. Thanks!